File: VB/WinWebSolution.Module/HighlightFocusedLayoutItemDetailViewControllerBase.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp

Namespace WinWebSolution.Module
	Public MustInherit Class HighlightFocusedLayoutItemDetailViewControllerBase
		Inherits ViewController(Of DetailView)
		Public Const HighlightFocusedLayoutItemAttributeName As String = "HighlightFocusedLayoutItem"
		Public Const EnableHighlightFocusedLayoutItemAttributeName As String = "EnableHighlightFocusedLayoutItem"
		Public Const ActiveKeyHighlightFocusedEditor As String = "HighlightFocusedLayoutItem"
		Protected Overrides Sub OnViewChanging(ByVal view As View)
			MyBase.OnViewChanging(view)
			Dim dv As DetailView = TryCast(view, DetailView)
			If dv IsNot Nothing Then
				Active(ActiveKeyHighlightFocusedEditor) = (CType(dv.Model, IModelDetailViewHighlightFocusedLayoutItem)).HighlightFocusedLayoutItem
			End If
		End Sub
		Protected MustOverride Sub ApplyFocusedStyle(ByVal element As Object)
	End Class
End Namespace

File: CS/WinWebSolution.Module/HighlightFocusedLayoutItemDetailViewControllerBase.cs
using DevExpress.ExpressApp;

namespace WinWebSolution.Module {
    public abstract class HighlightFocusedLayoutItemDetailViewControllerBase : ViewController<DetailView> {
        public const string HighlightFocusedLayoutItemAttributeName = "HighlightFocusedLayoutItem";
        public const string EnableHighlightFocusedLayoutItemAttributeName = "EnableHighlightFocusedLayoutItem";
        public const string ActiveKeyHighlightFocusedEditor = "HighlightFocusedLayoutItem";
        protected override void OnViewChanging(View view) {
            base.OnViewChanging(view);
            DetailView dv = view as DetailView;
            if (dv != null)
                Active[ActiveKeyHighlightFocusedEditor] = ((IModelDetailViewHighlightFocusedLayoutItem)dv.Model).HighlightFocusedLayoutItem;
        }
        protected abstract void ApplyFocusedStyle(object element);
    }
}